crates/sui-framework/sources/pay.move

```diff
@@ -0,0 +1,80 @@
+// Copyright (c) 2022, Mysten Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+/// This module provides handy functionality for wallets and `sui::Coin` management.
+module sui::pay {
+    use sui::tx_context::{Self, TxContext};
+    use sui::coin::{Self, Coin};
+    use sui::transfer;
+    use std::vector;
+
+    /// For when trying to split a coin more times than its balance allows.
+    const ENotEnough: u64 = 0;
+
+    /// For when invalid arguments are passed to a function.
+    const EInvalidArg: u64 = 1;
+
+    /// Split coin `self` to two coins, one with balance `split_amount`,
+    /// and the remaining balance is left is `self`.
+    public entry fun split<T>(self: &mut Coin<T>, split_amount: u64, ctx: &mut TxContext) {
+        transfer::transfer(
+            coin::split(self, split_amount, ctx),
+            tx_context::sender(ctx)
+        )
+    }
+
+    /// Split coin `self` into multiple coins, each with balance specified
+    /// in `split_amounts`. Remaining balance is left in `self`.
+    public entry fun split_vec<T>(
+        self: &mut Coin<T>, split_amounts: vector<u64>, ctx: &mut TxContext
+    ) {
+        let i = 0;
+        let len = vector::length(&split_amounts);
+        while (i < len) {
+            split(self, *vector::borrow(&split_amounts, i), ctx);
+            i = i + 1;
+        };
+    }
+
+    /// Split coin `self` into `n` coins with equal balances. If the balance is
+    /// not evenly divisible by `n`, the remainder is left in `self`. Return
+    /// newly created coins.
```

Comment:
If I understand the implementation of this function, it splits `self` into `n - 1` new coins with equal balance, and leaves the remainder in `self` (which could be equal to the other coins if its value was divisible by `n`, but it could also be more).

To me, the documentation currently suggests something different:  That it produces `n` new coins, and `self` gets `0` value if it all divides cleanly or the remainder i.e. strictly less than `coin::value(self) / n` otherwise.